crates/core/tedge_mapper/src/main.rs

```diff
@@ -75,24 +75,42 @@ impl fmt::Display for MapperName {
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    let mapper = MapperOpt::parse();
-    tedge_utils::logging::initialise_tracing_subscriber(mapper.debug);
+    let mapper_opt = MapperOpt::parse();
+    tedge_utils::logging::initialise_tracing_subscriber(mapper_opt.debug);
 
-    let component = lookup_component(&mapper.name);
+    let component = lookup_component(&mapper_opt.name);
 
     let tedge_config_location =
-        tedge_config::TEdgeConfigLocation::from_custom_root(&mapper.config_dir);
+        tedge_config::TEdgeConfigLocation::from_custom_root(&mapper_opt.config_dir);
     let config = tedge_config::TEdgeConfigRepository::new(tedge_config_location.clone()).load()?;
     // Run only one instance of a mapper
     let _flock = check_another_instance_is_not_running(
-        &mapper.name.to_string(),
+        &mapper_opt.name.to_string(),
         &config.query(RunPathDefaultSetting)?.into(),
     )?;
 
-    if mapper.init {
-        let mut mapper = CumulocityMapper::new();
-        mapper.init_session().await
-    } else if mapper.clear {
+    if mapper_opt.init {
+        match mapper_opt.name {
```

Comment:
This `match` is really suspicious. Please consider to remove it.

* We already have a `component` struct that is built after the `mapper_opt.name`. Why a new `mapper` struct built along the same lines ?
* The switch is on a name and that name is then used to call methods.
* Why hard-coded constants (e.g. `"az"`) that happen to be equals to the enums (e.g. `MapperName::Az`)?

Reply:
> This `match` is really suspicious. Please consider to remove it.
> 
> * We already have a `component` struct that is built after the `mapper_opt.name`. Why a new `mapper` struct built along the same lines ?

The `component` implements trait `TedgeComponent` trait that has only `start` method. Here `init` method is needed, so the `init` is created as part of the `mapper` struct`, that needs to be instantiated based on type of the mapper.
If I can add `init` and `clear` methods to this trait, then the code will look much simpler.

> * The switch is on a name and that name is then used to call methods.
> * Why hard-coded constants (e.g. `"az"`) that happen to be equals to the enums (e.g. `MapperName::Az`)?

fixed.